report_qweb_pdf_watermark/models/report.py

```diff
@@ -17,73 +17,66 @@
     from PIL import PdfImagePlugin  # noqa: F401
 except ImportError:
     pass
+try:
+    from papersize import parse_couple, SIZES
+except ImportError:
+    raise
 from openerp import api, models, tools
 logger = getLogger(__name__)
 
 
 class Report(models.Model):
     _inherit = 'report'
 
-    @api.v7
-    def get_pdf(
-        self, cr, uid, ids, report_name, html=None, data=None, context=None
-    ):
-        # pylint: disable=R8110
-        # this override cannot be done in v8 api
-        result = super(Report, self).get_pdf(
-            cr, uid, ids, report_name, html=html, data=data,
-            context=context
-        )
-        report = self._get_report_from_name(cr, uid, report_name)
-        watermark = None
+    @api.multi
+    def _scale_watermark(self, report, image):
+        size = parse_couple(
+            SIZES[report.paperformat_id.format.lower()])
+        size = int(size[0]), int(size[1])
+        return image.resize(size, resample=Image.LANCZOS)
```

Comment:
you actually stretch the watermark, which distorts proportions if the watermark doesn't happen to have the same proportions as the report. either make this configurable, or decide for using either the x or y side for scaling

Reply:
> you actually stretch the watermark, which distorts proportions if the watermark doesn't happen to have the same proportions as the report.

We have to be specific with terminology so that we can understand each other. The operation I apply here is `scaling` or `resizing`. Because the image file is PNG which is a raster graphics file format, when **scaling** there will be a loss of quality because the resolution stays the same.

By giving the option to `scale` here, I mean to bring the watermark to the same size (**not** resolution) as the already existing report.

I am already giving the opportunity to the user not to scale at https://github.com/OCA/reporting-engine/pull/283/files#diff-0f06fef7d1f6955c05bfff02a178967cR11. Should they choose this they should know the consequences (or they will quickly be able to see them).

Having said that, what troubles me is that while I understand that when you scale **up** an image there will be a loss of quality, when you scale **down** it should not, which is what happens here.

Reply:
400x400 -> 800x800: scaling
400x400 -> 600x800: stretching, distorting proportions and that's what you do here